tools/sloop.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ */
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Function Declarations
+protected void add_sloop_shadow( object ob, string str );
+protected void dest_sloop_shadows( object ob, string str );
+protected object * query_sloop_shadows( object ob, string str );
+void add_slooper_shadow( object ob );
+void add_sloopee_shadow( object ob );
+void dest_slooper_shadow( object ob );
+void dest_sloopee_shadow( object ob );
+object find_slooper_shadow( object ob );
+object find_sloopee_shadow( object ob );
+
+/**
+ * Function filters all shdow objects and returns a filtered array of shadows
+ * matching either SLOOPEE_SHAD or SLOOPER_SHAD
+ * @param  object ob   Object passed to check for shadow
+ * @param  string str    string to match (Either SLOOPER_SHAD or SLOOPEE_SHAD)
+ * @return   array of matching shadow(s)
+ */
+protected object * query_sloop_shadows( object ob, string str ) {
+    object * shadow_ob = shadow_list( ob );
+    return filter( shadow_ob, (: return ( program_name( $1 ) == $2 ); :), str );
```

Comment:
This is a good a place as any to bring up horizontal whitespace. I tend to be liberal with vertical whitespace and conservative with horizontal whitespace. What I can fit on a line matters more to me than what I can fit on a page, and 80 characters isn't much. That said, I would write this line like this:
```  return filter(shadow_ob, (: return (program_name($1) == $2); :), str);```

By sacrificing a small amount of readability, you go from 81 chars down to 73, and now you don't have to find a place to linebreak anymore. What I changed:
* indent width is now 2 chars instead of 4
* function parameters are no longer enclosed in whitespace
* no enclosing whitespace for parenthetical expressions

What I kept:
* flow control statements like _return_ still followed by whitespace
* operators like _==_ still enclosed by whitespace
* array/mapping/closure constructors _({ })_ _([ ])_ and  _(: :)_ still contain inner whitespace
* comma operator still followed by whitespace

These rules aren't set in stone (yet), so lemme know what you think of them.